packages/pg-pool/index.js

```diff
@@ -248,6 +251,19 @@ class Pool extends EventEmitter {
       } else {
         this.log('new client connected')
 
+        if (0 !== this.options.maxLifetimeSeconds) {
+          setTimeout(() => {
+            this.log('ending client due to expired lifetime')
+            this._expired.add(client)
+            if (this._idle.length) {
```

Comment:
Seems like the `-1` check already covers this.

Reply:
Yes. I was just thinking it was a quicker initial check. I'll remove it.